services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers_tasks.py

```diff
@@ -0,0 +1,317 @@
+    "/containers/ports/inputs:pull",
+    summary="Pull input ports data",
+    status_code=status.HTTP_202_ACCEPTED,
+    response_model=TaskId,
+)
+@cancel_on_disconnect
+async def ports_inputs_pull_task(
+    request: Request,
+    port_keys: Optional[list[str]] = None,
+    task_manager: TaskManager = Depends(get_task_manager),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+    mounted_volumes: MountedVolumes = Depends(get_mounted_volumes),
+) -> TaskId:
+    assert request  # nosec
+
+    try:
+        task_id = start_task(
+            task_manager=task_manager,
+            handler=task_ports_inputs_pull,
+            unique=True,
+            port_keys=port_keys,
+            mounted_volumes=mounted_volumes,
+            rabbitmq=rabbitmq,
+        )
+        return task_id
+    except TaskAlreadyRunningError as e:
+        raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@containers_router_tasks.post(
+    "/containers/ports/outputs:pull",
+    summary="Pull output ports data",
+    status_code=status.HTTP_202_ACCEPTED,
+    response_model=TaskId,
+)
+@cancel_on_disconnect
+async def ports_outputs_pull_task(
+    request: Request,
+    port_keys: Optional[list[str]] = None,
+    task_manager: TaskManager = Depends(get_task_manager),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+    mounted_volumes: MountedVolumes = Depends(get_mounted_volumes),
+) -> TaskId:
+    assert request  # nosec
+
+    try:
+        task_id = start_task(
+            task_manager=task_manager,
+            handler=task_ports_outputs_pull,
+            unique=True,
+            port_keys=port_keys,
+            mounted_volumes=mounted_volumes,
+            rabbitmq=rabbitmq,
+        )
+        return task_id
+    except TaskAlreadyRunningError as e:
+        raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@containers_router_tasks.post(
+    "/containers/ports/outputs:push",
+    summary="Push output ports data",
+    status_code=status.HTTP_202_ACCEPTED,
+    response_model=TaskId,
+)
+@cancel_on_disconnect
+async def ports_outputs_push_task(
+    request: Request,
+    port_keys: Optional[list[str]] = None,
+    task_manager: TaskManager = Depends(get_task_manager),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+    mounted_volumes: MountedVolumes = Depends(get_mounted_volumes),
+) -> TaskId:
+    assert request  # nosec
+
+    try:
+        task_id = start_task(
+            task_manager=task_manager,
+            handler=task_ports_outputs_push,
+            unique=True,
+            port_keys=port_keys,
+            mounted_volumes=mounted_volumes,
+            rabbitmq=rabbitmq,
+        )
+        return task_id
+    except TaskAlreadyRunningError as e:
+        raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@containers_router_tasks.post(
+    "/containers:restart",
+    summary="Removes the previously started service and returns the docker-compose output",
```

Comment:
this summary seems not totally correct right? you just return the task. is the output still returned? and actually why do I need it?

Reply:
I've corrected it. It just restarts the containers.